sklearn/tests/test_partial_dependence.py

```diff
@@ -0,0 +1,479 @@
+"""
+Testing for the partial dependence module.
+"""
+
+import numpy as np
+from numpy.testing import assert_array_almost_equal
+import pytest
+
+import sklearn
+from sklearn.utils.testing import assert_raises_regex
+from sklearn.utils.testing import assert_warns_message
+from sklearn.utils.testing import if_matplotlib
+from sklearn.partial_dependence import partial_dependence
+from sklearn.partial_dependence import plot_partial_dependence
+from sklearn.partial_dependence import _grid_from_X
+from sklearn.partial_dependence import _partial_dependence_brute
+from sklearn.partial_dependence import _partial_dependence_recursion
+from sklearn.ensemble import GradientBoostingClassifier
+from sklearn.ensemble import GradientBoostingRegressor
+from sklearn.linear_model import LinearRegression
+from sklearn.linear_model import LogisticRegression
+from sklearn.linear_model import MultiTaskLasso
+from sklearn.svm import SVC
+from sklearn.datasets import load_boston, load_iris
+from sklearn.datasets import make_classification, make_regression
+from sklearn.cluster import KMeans
+from sklearn.metrics import r2_score
+from sklearn.preprocessing import PolynomialFeatures
+from sklearn.dummy import DummyClassifier
+
+
+# toy sample
+X = [[-2, -1], [-1, -1], [-1, -2], [1, 1], [1, 2], [2, 1]]
+y = [-1, -1, -1, 1, 1, 1]
+
+
+def binary_classification():
+    # returns (X, y), n_targets  <-- as expected in the output of partial_dep()
+    return make_classification(random_state=0), 1
+
+
+def multiclass_classification():
+    # returns (X, y), n_targets  <-- as expected in the output of partial_dep()
+    return (make_classification(n_classes=3, n_clusters_per_class=1,
+                                random_state=0), 3)
+
+
+def regression():
+    # returns (X, y), n_targets  <-- as expected in the output of partial_dep()
+    return make_regression(random_state=0), 1
+
+
+def multioutput_regression():
+    # returns (X, y), n_targets  <-- as expected in the output of partial_dep()
+    return make_regression(n_targets=2, random_state=0), 2
+
+
+@pytest.mark.filterwarnings('ignore:Default solver will be changed ')  # 0.22
+@pytest.mark.filterwarnings('ignore:Default multi_class will be')  # 0.22
+@pytest.mark.parametrize('Estimator, method, data', [
+    (GradientBoostingClassifier, 'recursion', binary_classification()),
```

Comment:
Not sure why you re-create this data for each test rather than reusing the same instances.

Reply:
Because it's easier to encapsulate the `n_targets` variable. Do you think I should turn them into generators? Or declare the datasets at the top of the file?

Reply:
oh nevermind... I understand what you meant now